homeassistant/components/dio_chacon/cover.py

```diff
@@ -0,0 +1,164 @@
+"""Cover Platform for Dio Chacon REV-SHUTTER devices."""
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api import DIOChaconAPIClient
+from dio_chacon_wifi_api.const import DeviceTypeEnum, ShutterMoveEnum
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DioChaconData
+from .entity import DioChaconEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    dio_chacon_data: DioChaconData = config_entry.runtime_data
+
+    entities: list[DioChaconCover] = []
+    for device in dio_chacon_data.list_devices:
+        if device["type"] == DeviceTypeEnum.SHUTTER.value:
+            cover = DioChaconCover(dio_chacon_data.dio_chacon_client, device)
+            # Registers a callback to update the device state when the server sends an event
+            dio_chacon_data.dio_chacon_client.set_callback_device_state_by_device(
+                cover.target_id, cover.callback_device_state
+            )
+            entities.append(cover)
+
+    async_add_entities(entities)
+
+
+class DioChaconCover(DioChaconEntity, CoverEntity):
+    """Object for controlling a Dio Chacon cover."""
+
+    # To avoid having name twice
+    _attr_has_entity_name = False
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.STOP
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    def __init__(
+        self,
+        dio_chacon_client: DIOChaconAPIClient,
+        device: dict[str, Any],
+    ) -> None:
+        """Initialize the cover."""
+
+        _LOGGER.debug(
+            "Adding DIO Chacon SHUTTER Cover with id=%s, name=%s, openlevel=%s, movement=%s and connected=%s",
+            device["id"],
+            device["name"],
+            device["openlevel"],
+            device["movement"],
+            device["connected"],
+        )
+
+        super().__init__(
+            dio_chacon_client, device["id"], device["name"], device["model"]
+        )
+
+        self._attr_available = device["connected"]
+        self._attr_current_cover_position = device["openlevel"]
+        self._attr_is_closing = device["movement"] == ShutterMoveEnum.DOWN.value
+        self._attr_is_opening = device["movement"] == ShutterMoveEnum.UP.value
+        self._attr_is_closed = self._attr_current_cover_position == 0
```

Comment:
Instead of duplicating the attribute update, I think you could better create a `def _update_attr(self) -> None` which you can call at the end of the constructor

(comment continues to the next comment)